File: LeetCode/Easy/Find Pivot Index/Java/findPivotIndex.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        int[] pre = new int[n+1];
        pre[0] = 0;
        for (int i=1; i<=n; i++)
            pre[i] = pre[i-1] + nums[i-1];

        for (int i=1; i<=n; i++) {
            int left = pre[i-1];
            int right = pre[n]-pre[i-1]-nums[i-1];
            if (left == right) return i-1;
        }
        return -1;
    }
}

File: LeetCode/Easy/Find Pivot Index/C++/find_pivot_index.cc
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int left[nums.size()];
        int right[nums.size()];
        left[0] = nums[0];
        right[nums.size()-1] = nums[nums.size()-1];
        for (int i=1; i<nums.size(); i++) left[i] = left[i-1] + nums[i];
        for (int i=nums.size()-2; i>=0; i--) right[i] = right[i+1] + nums[i];
        for (int i=0; i<nums.size(); i++) if (left[i] == right[i]) return i;
        return -1;
    }
};

/**
[1, 7, 3, 6, 5, 6]
[1, 8, 11, 17, 22, 28]
[28 ,27 ,20 ,17 ,11 ,6]
*/
